src/dswx_sar/dswx_comparison.py

```diff
@@ -0,0 +1,205 @@
+                      'REFERENCE_WATER_SOURCE', 'SOFTWARE_VERSION']:
+                continue
+            if metadata_2[k1] != v1:
+                flag_same_metadata = False
+                metadata_error_message = (
+                    f'* contents of metadata key {k1} from'
+                    f' input 1 has value "{v1}" whereas the same key in'
+                    f' input 2 metadata has value "{metadata_2[k1]}"')
+                break
+    return metadata_error_message, flag_same_metadata
+
+def compare_dswx_sar_products(file_1, file_2):
+
+    if not os.path.isfile(file_1):
+        print(f'ERROR file not found: {file_1}')
+        return False
+
+    if not os.path.isfile(file_2):
+        print(f'ERROR file not found: {file_2}')
+        return False
+
+    print('Comparing files:')
+    print(f'    file 1: {file_1}')
+    print(f'    file 2: {file_2}')
+
+    flag_all_ok = [True]
+
+    # TODO: compare projections ds.GetProjection()
+    layer_gdal_dataset_1 = gdal.Open(file_1, gdal.GA_ReadOnly)
+    geotransform_1 = layer_gdal_dataset_1.GetGeoTransform()
+    projection_1 = layer_gdal_dataset_1.GetProjection()
+    metadata_1 = layer_gdal_dataset_1.GetMetadata()
+    nbands_1 = layer_gdal_dataset_1.RasterCount
+
+    layer_gdal_dataset_2 = gdal.Open(file_2, gdal.GA_ReadOnly)
+    geotransform_2 = layer_gdal_dataset_2.GetGeoTransform()
+    projection_2 = layer_gdal_dataset_2.GetProjection()
+    metadata_2 = layer_gdal_dataset_2.GetMetadata()
+    nbands_2 = layer_gdal_dataset_2.RasterCount
+
+    # compare number of bands
+    flag_same_nbands =  nbands_1 == nbands_2
+    flag_same_nbands_str = _get_prefix_str(flag_same_nbands, flag_all_ok)
+    prefix = ' ' * 7
+    print(f'{flag_same_nbands_str}Comparing number of bands')
+    if not flag_same_nbands:
+        print(prefix + f'Input 1 has {nbands_1} bands and input 2'
+              f' has {nbands_2} bands')
+        return False
+
+    # compare array values
+    print('Comparing DSWx bands...')
+    for b in range(1, nbands_1 + 1):
+        gdal_band_1 = layer_gdal_dataset_1.GetRasterBand(b)
+        gdal_band_2 = layer_gdal_dataset_2.GetRasterBand(b)
+        image_1 = gdal_band_1.ReadAsArray()
+        image_2 = gdal_band_2.ReadAsArray()
+        flag_bands_are_equal = np.allclose(
+            image_1, image_2, atol = COMPARE_DSWX_SAR_PRODUCTS_ERROR_TOLERANCE,
+            equal_nan=True)
+        flag_bands_are_equal_str = _get_prefix_str(flag_bands_are_equal,
+                                                   flag_all_ok)
+        print(f'{flag_bands_are_equal_str}     Band {b} -'
+              f' {gdal_band_1.GetDescription()}"')
+        if not flag_bands_are_equal:
+            _print_first_value_diff(image_1, image_2, prefix)
+
+    # compare geotransforms
+    flag_same_geotransforms = np.array_equal(geotransform_1, geotransform_2)
+    flag_same_geotransforms_str = _get_prefix_str(flag_same_geotransforms,
```

Comment:
Please think about if printing out the geotransfmrm values would be helpful when `flag_same_geotransforms` is `False`

Reply:
This is error message when geotranform, projection and contents are different. The geotransform is now printed out.
```
Comparing files:
    file 1: OPERA_L3_DSWx-S1_T43RBH_20211012T011016Z_20230813T052057Z_vNone_B01_WTR.tif
    file 2: OPERA_L3_DSWx-S1_T18MYU_20200702T231843Z_20230522T165954Z_v0.1_B01_WTR.tif
[OK]   Comparing number of bands
Comparing DSWx bands...
[FAIL]      Band 1 - Water classification (WTR)"
            * input 1 has value "255" in position (x: 0, y: 516) whereas input 2 has value "0" in the same position.
[FAIL] Comparing geotransform
       * input 1 geotransform with content "(200000.0, 30.0, 0.0, 2809800.0, 0.0, -30.0)" differs from input 2 geotransform with content "(700000.0, 30.0, 0.0, 9409800.0, 0.0, -30.0)".
[FAIL] Comparing projection
       * input 1 projection with content "PROJCS["WGS 84 / UTM zone 43N",GEOGCS["WGS 84",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9122"]],AUTHORITY["EPSG","4326"]],PROJECTION["Transverse_Mercator"],PARAMETER["latitude_of_origin",0],PARAMETER["central_meridian",75],PARAMETER["scale_factor",0.9996],PARAMETER["false_easting",500000],PARAMETER["false_northing",0],UNIT["metre",1,AUTHORITY["EPSG","9001"]],AXIS["Easting",EAST],AXIS["Northing",NORTH],AUTHORITY["EPSG","32643"]]" differs from input 2 projection with content "PROJCS["WGS 84 / UTM zone 18S",GEOGCS["WGS 84",DATUM["WGS_1984",SPHEROID["WGS 84",6378137,298.257223563,AUTHORITY["EPSG","7030"]],AUTHORITY["EPSG","6326"]],PRIMEM["Greenwich",0,AUTHORITY["EPSG","8901"]],UNIT["degree",0.0174532925199433,AUTHORITY["EPSG","9122"]],AUTHORITY["EPSG","4326"]],PROJECTION["Transverse_Mercator"],PARAMETER["latitude_of_origin",0],PARAMETER["central_meridian",-75],PARAMETER["scale_factor",0.9996],PARAMETER["false_easting",500000],PARAMETER["false_northing",10000000],UNIT["metre",1,AUTHORITY["EPSG","9001"]],AXIS["Easting",EAST],AXIS["Northing",NORTH],AUTHORITY["EPSG","32718"]]".
[OK]   Comparing metadata
```